packages/backend/src/SynthqlError.ts

```diff
@@ -135,6 +135,18 @@ export class SynthqlError extends Error {
 
         return new SynthqlError(new Error(), type, lines.join('\n'), 404);
     }
+
+    static createPermissionsError() {
+        const type = 'PermissionsError';
+
+        const lines = [
+            'A query with a permissions list (ACL) included,',
+            'is missing matching permissions in the',
+            'context object permissions list',
+        ];
+
+        return new SynthqlError(new Error(), type, lines.join('\n'), 404);
```

Comment:
404 is not the right error code here, it's 403

Reply:
> To summarize the main difference between the two, although both status codes represent access denial, 401 errors address authentication issues, and 403 errors point towards authorization problems. It is important to make this distinction as incorrectly handling these responses can leave you susceptible to exploits from malicious attackers.

Source: https://supertokens.com/blog/http-error-codes-401-vs-403#:~:text=Differences%20between%20error%20codes%20401%20and%20403,-While%20both%20HTTP&text=Authorization%3A%20A%20401%20error%20focuses,valid%20authentication%20but%20inadequate%20authorization.